src/span.cpp

```diff
@@ -300,20 +300,28 @@ struct VariantVisitor {
 };
 }  // namespace
 
+// Replaces some characters in span tag keys with a '.'
+// For now, only ':' is replaced.
+std::string replacePunctuation(std::string tag) {
```

Comment:
Just for visibility, this is what we do in `dd-trace-py`, which is based on the tags definition: https://docs.datadoghq.com/tagging/#defining-tags

https://github.com/DataDog/dd-trace-py/blob/1db7121016a130bb8535be2f1471ffa965cc727a/ddtrace/http/headers.py#L11-L14


(read as an FYI, not requesting a change here)

Reply:
Good to know. `normalizeTagName` would be a better name. Since it affects creating facets, changing it in the future would mean users need to reconfigure things.
I'll think a bit more before choosing to update or merge this.